Assets/Scripts/Collectable.cs

```diff
@@ -7,16 +7,33 @@ public class Collectable : MonoBehaviour {
     [SerializeField]
     private float rotationSpeed;
 
-	private void FixedUpdate ()
+    public static float collectables = 0f;
```

Comment:
One thing I forgot to talk about in my video is that static values hang around until the game is turned off. So you need to manually reset them if needed.

- If you die and the level reloads, do you need to manually reset this back to the previous value?
- If you move to the next level, should this reset to zero or continue counting from where you left off?
- I think when you respawn from a checkpoint it's OK to keep the value, because those collectables will still be removed and this is a friendly to the player game design decision.

Reply:
Fixed in PlayerController Respawn function and CollectableManager.  CollectableManager sets public float startingCollectables equal to Collectable.collectables at Start.  This should only be at the start of each Scene.  PlayerController has a SerializeField for collectableUI, which then is called during Respawn function to set Collectable.collectables equal to startingCollectables.

This may have been a convoluted way to do this.